std/datetime/date.d

```diff
@@ -9689,6 +9691,16 @@ if (units == "days")
     assert(!valid!"days"(2017, 2, 29));
 }
 
+private short castToYear(int year, string file = __FILE__, size_t line = __LINE__) @safe pure
+{
+    import std.conv : to, ConvOverflowException;
+    import std.format : format;
+
+    try
+        return year.to!short;
+    catch (ConvOverflowException)
+        throw new DateTimeException(format("year %s doesn't fit to Date.", year), file, line);
```

Comment:
It's my understanding that the try-catch adds overhead. Given how often this code is going to be used with date/time stuff, it would make more sense to just check the bounds and throw rather than using `to`.

Reply:
try-catch for `.to` isn't adds overhead here: if we just check bounds manually then create `DateTimeException` manually it will be same cost

Overhead here is added only by cost of the second exception creation if only bounds check failed. For for `year` it is very rare case, but this makes code more common and clear.

So, since we're using exceptions in this module to report errors that's fine.